src/Mobs/PassiveMonster.cpp

```diff
@@ -43,25 +73,122 @@ void cPassiveMonster::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 	{
 		CheckEventLostPlayer();
 	}
-	cItems FollowedItems;
-	GetFollowedItems(FollowedItems);
-	if (FollowedItems.Size() <= 0)
+	if ((m_LovePartner != nullptr) && m_LovePartner->IsDestroyed())
 	{
-		return;
+		m_LovePartner = nullptr;
 	}
-	cPlayer * a_Closest_Player = m_World->FindClosestPlayer(GetPosition(), static_cast<float>(m_SightDistance));
-	if (a_Closest_Player != nullptr)
+	if (m_LovePartner != nullptr)
 	{
-		cItem EquippedItem = a_Closest_Player->GetEquippedItem();
-		if (FollowedItems.ContainsType(EquippedItem))
+		// if we have a partner, bump into them until baby is made
+		if (m_MatingTimer > 0)
 		{
-			Vector3d PlayerPos = a_Closest_Player->GetPosition();
-			MoveToPosition(PlayerPos);
+			Vector3d Pos = m_LovePartner->GetPosition();
+			MoveToPosition(Pos);
+		}
+		else
+		{
+			// spawn baby
+			Vector3f Pos = (GetPosition() + m_LovePartner->GetPosition()) * 0.5;
+			m_World->SpawnMob(Pos.x, Pos.y, Pos.z, GetMobType(), true);
+
+			cFastRandom Random;
+			m_World->SpawnExperienceOrb(Pos.x, Pos.y, Pos.z, 1 + Random.NextInt(6));
+
+			m_LovePartner->ResetLoveMode();
+			ResetLoveMode();
 		}
 	}
+	else
+	{
+		cItems FollowedItems;
+		GetFollowedItems(FollowedItems);
+		if (FollowedItems.Size() > 0)
+		{
+			cPlayer * a_Closest_Player = m_World->FindClosestPlayer(GetPosition(), static_cast<float>(m_SightDistance));
```

Comment:
What about a DoWithClosestPlayers(Position, Callback)?
It would sort the players based on how far they are away from the position. The closest player first, and the furthest as last. When the callback returns true it stops.